lambdas/data-migration2/src/executions.ts

```diff
@@ -80,34 +121,26 @@ export const migrateExecutions = async (
     skipped: 0,
   };
 
-  let record = await searchQueue.peek();
-  /* eslint-disable no-await-in-loop */
-  while (record) {
-    migrationResult.total_dynamo_db_records += 1;
-
-    if (migrationResult.total_dynamo_db_records % loggingInterval === 0) {
-      logger.info(`Batch of ${loggingInterval} execution records processed, ${migrationResult.total_dynamo_db_records} total`);
-    }
-
-    try {
-      await migrateExecutionRecord(<ExecutionRecord>record, knex);
-      migrationResult.migrated += 1;
-    } catch (error) {
-      if (error instanceof RecordAlreadyMigrated) {
-        migrationResult.skipped += 1;
-      } else {
-        migrationResult.failed += 1;
-        logger.error(
-          `Could not create execution record in RDS for Dynamo execution arn ${record.arn}:`,
-          error
-        );
-      }
-    }
+  const totalSegments = executionMigrationParams.parallelScanSegments ?? 5;
+
+  logger.info(`Starting parallel scan of executions with ${totalSegments} parallel segments`);
+
+  await parallelScan(
+    totalSegments,
+    {
+      TableName: executionsTable,
+      Limit: executionMigrationParams.parallelScanLimit,
+    },
+    (items) => processExecutionItems(
+      items,
+      migrationResult,
+      knex,
+      loggingInterval,
+      executionMigrationParams.writeConcurrency
+    )
+  );
 
-    await searchQueue.shift();
-    record = await searchQueue.peek();
-  }
-  /* eslint-enable no-await-in-loop */
-  logger.info(`successfully migrated ${migrationResult.migrated} execution records`);
+  logger.info(`Finished parallel scan of executions with ${totalSegments} parallel segments.`);
+  logger.info(`successfully migrated ${migrationResult.migrated} out of ${migrationResult.total_dynamo_db_records} execution records`);
```

Comment:
Ah, I see, it's because the typing isn't consistent between the tables.    Short of a more complex (and therefore less Good(tm))  return + parse/compute setup that probably can't be avoided.